verbose: 4
num_drone: 8
num_obs: 16  # ignored if read_config is false



# -------- qpoases to be added-------------
solver: "quadprog" # qpoases or quadprog

# @@ True = on_demand, False = continuous
on_demand: true      


# @@ Optimizer parameters
world:      3   
num:        30
t_plan:     3
num_up:     300
dist_stop:  0.2
free_space: true
kappa:      15

max_time:   50

order_smoothness: 4
weight_goal:       7000
weight_smoothness: 100
weight_quad_slack: 50
weight_lin_slack: 50000


# @@ Crazflie Configuration
# true kinematic values
vel_max: 1.73

# scaled thrust values 
use_thrust_values: true
f_max: 1.5    # 1.5g
f_min: 0.3    # 0.3g
gravity: 9.8

acc_max: 2.0


a_drone: 0.075
b_drone: 0.075
c_drone: 0.24

prox_agent: 0.2
prox_obs: 0.2

buffer:  0.06


# @@ Room limits
x_lim: [-2.0, 2.0]
y_lim: [-2.0, 2.0]
z_lim: [+0.2, 2.2]


# @@ Obstacle Configuration 
pos_static_obs: [[-1.359,  1.535,  0.   ],
                [-0.708,  1.038,  0.   ],
                [-0.046,  1.452,  0.   ],
                [ 0.424,  1.006,  0.   ],
                [ 0.969,  1.484,  0.   ],
                [-1.146,  0.37 ,  0.   ],
                [-0.484,  0.285,  0.   ],
                [-0.003, -0.128,  0.   ],
                [ 0.744,  0.37 ,  0.   ],
                [ 1.139, -0.309,  0.   ],
                [ 0.84 , -1.03 ,  0.   ],
                [ 0.029, -0.829,  0.   ],
                [-0.644, -0.701,  0.   ],
                [-1.135, -1.179,  0.   ],
                [-1.349, -0.394,  0.   ],
                [-0.441, -1.518,  0.   ],
                [ 0.264, -1.56 ,  0.   ],
                [ 1.31 , -1.497,  0.   ],
                [-1.562, -1.624,  0.   ],
                [-1.53 ,  0.837,  0.   ]]

dim_static_obs: [[+0.13, +0.13, 10.0],            
                [+0.13, +0.13, 10.0],             
                [+0.13, +0.13, 10.0],             
                [+0.13, +0.13, 10.0],             
                [+0.13, +0.13, 10.0],            
                [+0.13, +0.13, 10.0],             
                [+0.13, +0.13, 10.0],             
                [+0.13, +0.13, 10.0],             
                [+0.13, +0.13, 10.0],             
                [+0.13, +0.13, 10.0],             
                [+0.13, +0.13, 10.0],              
                [+0.13, +0.13, 10.0],              
                [+0.13, +0.13, 10.0], 
                [+0.13, +0.13, 10.0], 
                [+0.13, +0.13, 10.0], 
                [+0.13, +0.13, 10.0], 
                [+0.13, +0.13, 10.0], 
                [+0.13, +0.13, 10.0], 
                [+0.13, +0.13, 10.0], 
                [+0.13, +0.13, 10.0]] 

# @ Start-Goal Configuration                
# init_drone: [[-1.0, -1.0, 0.5],
#              [+1.0, +1.0, 0.5],
#              [+1.0, -1.0, 0.5],
#              [-1.0, +1.0, 0.5],
#              [+1.0, +0.0, 0.5],
#              [-1.0, +0.0, 0.5],
#              [+0.0, +1.0, 0.5],
#              [-0.0, -1.0, 0.5]]

# goal_drone: [[+1.0, +1.0, 0.5],
#              [-1.0, -1.0, 0.5],
#              [-1.0, +1.0, 0.5],
#              [+1.0, -1.0, 0.5],
#              [-1.0, +0.0, 0.5],
#              [+1.0, +0.0, 0.5],
#              [+0.0, -1.0, 0.5],
#              [-0.0, +1.0, 0.5]]

init_drone: [[-0.0, -0.0, 0.5],
          [-1.5, -1.5, 0.5],
          [+1.5, +1.5, 0.5],
          [+1.5, -1.5, 0.5],
          [-1.5, +1.5, 0.5],
          [+1.5, +0.0, 0.5],
          [-1.5, +0.0, 0.5],
          [+0.0, +1.5, 0.5],
          [-0.0, -1.5, 0.5],
          [-1.5, -0.75, 0.5],
          [+1.5, +0.75, 0.5],
          [+1.5, -0.75, 0.5],
          [-1.5, +0.75, 0.5],
          [-0.75, -1.5, 0.5],
          [+0.75, +1.5, 0.5]]

goal_drone: [[-0.0, -0.0, 0.5],     #1
          [+1.5, +1.5, 0.5],    #2
          [-1.5, -1.5, 0.5],    #3
          [-1.5, +1.5, 0.5],    #4
          [+1.5, -1.5, 0.5],    #5
          [-1.5, +0.0, 0.5],    #6
          [+1.5, -0.0, 0.5],    #7
          [-0.0, -1.5, 0.5],    #8
          [+0.0, +1.5, 0.5],    #9
          [+1.5, +0.75, 0.5],   #10
          [-1.5, -0.75, 0.5],   #11
          [-1.5, +0.75, 0.5],   #12
          [+1.5, -0.75, 0.5],   #13
          [+0.75, +1.5, 0.5],
          [-0.75, -1.5, 0.5]]   #1